### ISSUE TITLE
SuppressWarnings("all") on Lombok-generated methods makes Checker ignore actual errors

### ISSUE THREAD

**Issue by PawelLipski** [kelloggm/checkerframework-gradle-plugin#85]
Let's consider the following scenario:

```java
interface IFoo {
  String getFoo(); // @NonNull by default
}

.... 

class Foo implements IFoo {
  @Getter
  @Nullable // overridden getter must NOT be nullable since it's declared non-null in IFoo
  private final String foo;
}
```

The generated getter will be something like:

```java
  @Nullable
  @SuppressWarnings("all")
  @lombok.Generated
  public String getFoo() {
    return this.foo;
  }
```

**If it wasn't for `@SuppressWarnings("all")`, then NullnessChecker would issue an error (which it definitely should in this case).**

Is it a (kinda-serious) bug or am I missing something?

Note: `org.checkerframework.gradle.plugin.CheckerFrameworkPlugin#handleLombokPlugin` has the following piece:

```groovy
// Also re-add suppress warnings annotations so that we don't get warnings from generated
// code.
delombokTask.format.put('suppressWarnings', 'generate')
```

**Comment by kelloggm** [comment]
> Note: org.checkerframework.gradle.plugin.CheckerFrameworkPlugin#handleLombokPlugin has the following piece:

This code exists to match Lombok's standard behavior, because normally users can't fix any warnings in Lombok-generated code. I don't think we can remove it without causing other kinds of warnings to show up in Lombok-generated code that normally would be suppressed.

That said, you're right that it's problematic in cases like the one you mentioned. I'm not sure how we'd be sure to expose that error without changing Lombok's behavior with respect to other warnings.

Overall, both the plugin and the CF itself treat code generated by Lombok as unchecked code (that is, like bytecode compiled elsewhere, not like source code).

If it would be helpful to you, I'd be happy to add an option to the plugin that disables the suppress warnings annotation.

**Comment by PawelLipski** [comment]
Yes, I'd be glad if such an option existed in the plugin, let's see how that goes (and if the other, now unsuppressed, warnings are going to be troublesome for us indeed).

**Comment by PawelLipski** [comment]
Thank you for quick response!

**Comment by PawelLipski** [comment]
By the way, are you aware of any way to influence the behavior of Lombok's annotation processor when your plugin is not used? Because IIUC Lombok also generates this `@SuppressWarnings` even when simply run as Gradle `annotationProcessor group: 'org.projectlombok', name: 'lombok', version: '1.18.12'`. This very issue is also present when Lombok annotation processor but not Lombok Gradle plugin is used -  Checker also wouldn't issue a warning for `Foo#foo` under such circumstances.

**Comment by kelloggm** [comment]
#86 has the configuration key. I've already released version `0.4.15`, so you should be able to try it out immediately without waiting for #86 to be merged. Please let me know how it goes!

> By the way, are you aware of any way to influence the behavior of Lombok's annotation processor when your plugin is not used?

Not other than delombok'ing the code without setting the flag to generate suppress warnings, and then compiling that, unfortunately.

**Comment by PawelLipski** [comment]
Thanks! I just tested 0.4.15, seems to works perfectly for our use cases :)

**Comment by PawelLipski** [comment]
Only requires `lombok.addNullAnnotations = checkerframework` in lomok.config to be present, otherwise Lombok-generated `equals` are broken.

**Comment by PawelLipski** [comment]
@kelloggm if you could also upvote a related issue https://github.com/rzwitserloot/lombok/issues/2433
